### DISCUSSION TITLE
"treat as single object" for a collection in multi tenant environment

### DISCUSSION THREAD

**Discussion by xu4wang** [directus/directus#12294] [Archive]
I'm prototyping a multi tenant app. following #9682 , approach 2.
I added tenant id directly on the user object which shared by multiple tenants.

So far so good, however, for some collections, it's actually single object for one tenant. for example
A collection named "Profiles", which stores the profile for each tenants.. One record for one tenant.

It will be good, if the tenant can have a UI similar to the native  "treat as single object" collection when modifying its profile.

Is there an easy way to achieve this?

I know I can move to multi projects approach but I am trying to stay with one project.

**Comment by rijkvanzanten** [comment]
Directus currently isn't built to handle multiple different items in its "single object" mode. I'm also curious how that would work for roles that have permissions to more than one of those items 🤔 Should the platform treat it as a regular collection for roles that have permissions to more than 1 (like admin users), and a "single item" for roles that don't?

**Reply by xu4wang** [reply]
Just a rough idea...

In the database, there is actually multiple items in the table "Profiles".  For some roles, they can only access one item,  so they should use a "single object" UI, say,  user can only read/modify the first object  in the collection it can read.

For other normal user, still fallback to normal table/map/cards/calendar view.

"treat as single object" will not be a checkbox for the collection, but accept a role list as value.

**Reply by rijkvanzanten** [reply]
> "treat as single object" will not be a checkbox for the collection, but accept a role list as value.

Oh that's interesting. I would've assumed it would be configured through permissions, and that the app UI would use a single page if it detects only a single item exists 🤔

**Reply by xu4wang** [reply]
another idea 

When the current user can see ONLY one item and cannot create/delete. The app just show the single page for that item.

Simple, no need any configuration.
